src/hotspot/os_cpu/linux_arm/atomic_linux_arm.hpp

```diff
@@ -30,6 +30,40 @@
 
 // Implementation of class atomic
 
+struct arm_atomic_funcs {
+  int64_t (*cmpxchg_long_func)(int64_t compare_value,
+                               int64_t exchange_value,
+                               volatile int64_t *dest);
+
+  int64_t (*load_long_func)(const volatile int64_t*);
+
+  void (*store_long_func)(int64_t, volatile int64_t*);
+
+  int32_t  (*add_func)(int32_t add_value, volatile int32_t *dest);
+
+  int32_t  (*xchg_func)(int32_t exchange_value, volatile int32_t *dest);
+
+  int32_t  (*cmpxchg_func)(int32_t compare_value,
+                           int32_t exchange_value,
+                           volatile int32_t *dest);
+
+  static int64_t cmpxchg_long_bootstrap(int64_t, int64_t, volatile int64_t*);
+
+  static int64_t load_long_bootstrap(const volatile int64_t*);
+
+  static void store_long_bootstrap(int64_t, volatile int64_t*);
+
+  static int32_t  add_bootstrap(int32_t add_value, volatile int32_t *dest);
+
+  static int32_t  xchg_bootstrap(int32_t exchange_value, volatile int32_t *dest);
+
+  static int32_t  cmpxchg_bootstrap(int32_t compare_value,
+                                    int32_t exchange_value,
+                                    volatile int32_t *dest);
+};
+
+extern arm_atomic_funcs _arm_atomic;
```

Comment:
Why is arm_atomic_funcs not an AllStatic class?  (And it has a non-conventional name.)